prow/kube/config.go

```diff
@@ -41,6 +41,7 @@ func LoadClusterConfigs(kubeconfig, buildCluster string) (configurations map[str
 		logrus.Warnf("Failed to create in-cluster config: %v", err)
 	} else {
 		defCtx = new(string)
+		*defCtx = DefaultClusterAlias
 		logrus.Info("* in-cluster")
 		configs[*defCtx] = *localCfg
```

Comment:
IMO both `configs[DefaultClusteralias]` and `configs[""]` should get set to `*localCfg`

At a minimum we should open an issue to unit test LoadClusterConfigs and add a TODO to address it somewhere in this code (which is what Steve is correctly getting at).

Reply:
I set both to `*localCfg` and created this issue for unit testing `LoadClusterConfigs` https://github.com/kubernetes/test-infra/issues/11620. 

In the current implementation is difficult to mock the dependencies.